### ISSUE TITLE
Edition / Description des attributs

### ISSUE THREAD

**Issue by stephyritz** [SPW-DIG/metawal-core-geonetwork#528]
Dans la continuité du travail fait et mentionné ici https://github.com/SPW-DIG/metawal-core-geonetwork/issues/506

En discutant en interne, il y aurait un intérêt pour nous de proposer une description des attributs systématiques de nos données stockées en DB. Nous commençons à tester un export systématique en json directement depuis oracle possédant plusieurs informations concernant les attributs. Avec sûrement la possibilité de rajouter une "définition" comme ce qui "attendu" dans Metawal. L'idée serait alors de pouvoir proposer de "charger" le catalogue d'attributs directement depuis ce json. Son url serait fixe et correspondrait à ceci : 
http://geoservices.test.wallonie.be/geotraitement/spwdatadownload/results/305e8131-1606-4c54-8f8b-6d48eec6678b/DATA_MODEL.json 

J'imagine que, pour chercher les infos dans le service REST, nous utilisions également le json ? Du coup, ça serait un peu la même procédure mais cette fois en allant chercher le json directement depuis une url bien définie. L'avantage serait de stocker ce json au même endroit que la donnée zippée préparée pour le téléchargement et que les informations soient directement tirées de la DB. De plus, à terme, cela pourrait faciliter le datalinking, cette notion qui commence à faire son bonhomme de chemin ici. Qu'en pensez-vous ?

**Comment by fxprunayre** [comment]
> Qu'en pensez-vous ?

Pourquoi ne pas générer un catalogue d'attributs en ISO19110 directement ? c'est pas bien plus compliqué qu'un JSON. Si on est obligé d'avoir du JSON, un JSON un brin standard ce serait bien ? eg. OpenAPI (https://swagger.io/docs/specification/data-models/) peut être du côté de OGC API feature voire ESRI Rest API si on en a pas mieux. Ca éviterait d'implémenter un truc spécifique MW.

**Comment by stephyritz** [comment]
Merci pour tes conseils. Je transfère à la personne qui s'en occupe. Je pense que le JSON est utilisé car c'est un format que les autres applications (par ex. formulaire dématérialisé pour les permis d'urbanisme) sont capables de comprendre facilement. Mais je ne pilote pas du tout ça donc là c’est mon interprétation. Je comprends l'intérêt de la création d'un JSON standard. Par contre, je ne maitrise pas du tout les technologies autour des API actuellement et ne peut malheureusement pas prendre le temps de m'y mettre en ce moment. Donc je passe l'info et te tiens au courant sur le sujet.

**Comment by stephyritz** [comment]
J'ai regardé à nouveau côté catalogue d'attribut en ISO19110. Je ne comprends pas trop : cette ISO est intégrée dans GN  --> https://geonetwork-opensource.org/manuals/trunk/ge/annexes/standards/iso19110.html
A quel endroit dans MW ?

Je peux éventuellement en créer un exemple à travers Metawal puis éventuellement utiliser FME pour populer un xml structuré de la sorte à partir des infos en DB. Voir même directement écrire dans Metawal à travers un CSW-T ou l'API (?!)

Cette procédure peut être faite en parallèle du JSON. Quand tu dis un brin standard, c'est par rapport, par exemple aux valeurs liées au type ? Ou bien plutôt dans la structure générale ?

**Comment by fxprunayre** [comment]
L'ISO19110 est un type de standard dans GN. Avec 19115-3 c'est inclu dedans - onglet contenu.


> Cette procédure peut être faite en parallèle du JSON.

Faut peut être éviter de se disperser aussi.



> Quand tu dis un brin standard, c'est par rapport, par exemple au valeur liée au type ? Ou bien plutôt dans la structure générale ?

La structure générale:
* une classe
* des propriétés
* des types
* des valeurs possibles
* voire des relations, des héritages ... tout ça est défini dans 19110.

**Comment by stephyritz** [comment]
Ok merci pour les infos. Répondre à plusieurs besoins n'est pas forcément se disperser. Et vu le nombre de projets à l'échelle de la région, je commence à comprendre que parfois les adaptations prennent du temps... Ceci dit, si les applications tierces arrivent à toutes interpréter le xml façon ISO19110, tant mieux. 

Concernant la structure générale du json, je pose la question parce que, bien que ne respectant pas les standards vis à vis de l'ISO19110, il est tout de même valide et donc d'une certaine manière, respecte un standard dans sa forme. D'où les interrogations de mon collègue qui, comme moi, n'est pas un spécialiste du catalogage.

**Comment by fxprunayre** [comment]
> respecte un standard dans sa forme.

Oui, après on a fait un import du format ESRI JSON vers 115-3, si on a le même, ça évite de refaire un mapping. C'est pas non plus super long. Mais si spécifique metawal, on ne le mettra pas dans GN.
